test/test_return.py

```diff
@@ -161,6 +161,14 @@ def test_return_location_edit(self):
         ), 'location method not allowed'
         assert self.get()['headers']['Location'] == 'blah'
 
+        assert 'success' in self.conf(
+            {"return": 302, "location": "foo${host}baz"}, 'routes/0/action'
+        ), 'configure location with a variable'
+        assert (
+            self.get(headers={"Host": "bar", "Connection": "close"}
+            )['headers']['Location'] == 'foobarbaz'
+        ), 'location with a variable'
+
     def test_return_invalid(self):
```

Comment:
I would suggest to add tests with 1) variable + encoding 2) something like `#...$var...` where `$var` contains `#?` symbols (to check that `#?` will be encoded).